Library/Homebrew/cmd/deps.rb

```diff
@@ -157,8 +160,11 @@ def deps_for_formula(f, recursive = false)
         if req.recommended?
           Requirement.prune if ignores.include?("recommended?") || dependent.build.without?(req)
         elsif req.test?
-          next if includes.include?("test?")
-          Requirement.prune
+          if includes.include?("test?")
+            Requirement.keep_but_prune_recursive_deps
```

Comment:
I think `Requirement` doesn't have `keep_but_prune_recursive_deps`

Reply:
`f.recursive_requirements` would just return `Requirement.expand(self)`:

https://github.com/Homebrew/brew/blob/144356e23846ca5332891b358d2a0a5789546dbe/Library/Homebrew/software_spec.rb#L196-L198

Which would just go over `f.recursive_dependencies` and "work" with `requirements` for each formula:

https://github.com/Homebrew/brew/blob/144356e23846ca5332891b358d2a0a5789546dbe/Library/Homebrew/requirement.rb#L191-L205

So, since all recursive deps are pruned already in previous step, their requirements shouldn't be "reached", I assume.

Reply:
Thanks @stek29!